### PR TITLE
rf#129 ecommerce trackers

### PR DESCRIPTION
**Pull request by artemiy312**
Closes fidals/refarm-site#129

See product's detail: 
default

Add to cart:
default

Remove from cart:
default

Submit of purchase:
default

### CODE REVIEW

File: front/js/components/order.es6 [hunk 1]
```diff
@@ -46,13 +47,26 @@
       'onCartUpdate', renderTable, fillSavedInputs,
       touchSpinReinit, restoreSelectedPayment, cityAutocomplete,
     );
-    $(DOM.fullForm).submit(() => mediator.publish('onOrderSend'));
+    $(DOM.fullForm).submit(() => {
+      let productsData = $(DOM.productRows).map((_, el) => {
```
Comment (duker33) [inline front/js/components/order.es6]: let's move it to `get_product_data` function or smth like this

File: front/js/shared/yandex.es6 [hunk 2]
```diff
@@ -19,24 +18,48 @@
...
+  // Load ecommerce plugin for gaTracker
+  ga('require', 'ecommerce');
+
+  let yaTracker = new YATracker(window.dataLayer, 'RUB');
+  let gaTracker = new GATracker(ga, 'ecommerce');
+
   const init = () => {
     setUpListeners();
+    // e-commerce container
+    // https://yandex.ru/support/metrika/data/e-commerce.html
```
Comment (duker33) [inline front/js/shared/yandex.es6]: it's not clear. What e-commerce container are doing here?

File: front/js/shared/yandex.es6 [hunk 3]
```diff
@@ -19,24 +18,48 @@
...
       reachGoal('FAST_BUY_SEND');
     });
-    mediator.subscribe('onOrderSend', () => {
+    mediator.subscribe('onOrderSend', (_, products) => {
       reachGoal('CMN_BUY_SEND');
       reachGoal('FULL_BUY_SEND');
+      // Use a dummy order's id, because we do not wait complete processing of
+      // purchase request.
+      yaTracker.purchase(products, {id: 'DummyId'});
+      gaTracker.purchase({id: 'DummyId'}, products);
```
Comment (duker33) [inline front/js/shared/yandex.es6]: diff args order. Looks like mistake

File: front/js/shared/yandex.es6 [hunk 4]
```diff
@@ -19,24 +18,48 @@
...
+    });
+    mediator.subscribe('onProductAdd', (_, id, count) => {
+      yaTracker.add([{id: id, quantity: count}]);
```
Comment (duker33) [inline front/js/shared/yandex.es6]: we should reach goal on product_add.
We have two different goals `PUT_IN_CART_FROM_PRODUCT` and `PUT_IN_CART_FROM_CATEGORY`.

If some diff code already do it, leave comment here plz

File: templates/layout/google_tag_manager.html [hunk 5]
```diff
@@ -24,6 +24,36 @@
 {% else %}
   {% if is_head_tag %}
     <script>
+      var gaObj = (function() {
```
Comment (duker33) [inline templates/layout/google_tag_manager.html]: `gaContext` might be more good name. Decide

File: front/js/shared/yandex.es6 (hunk 3 above)
Comment (artemiy312) [reply front/js/shared/yandex.es6]: @duker33 It is an issue of refarm-site [ya](https://github.com/fidals/refarm-site/blob/master/ecommerce/front/trackers.es6#L30) [ga](https://github.com/fidals/refarm-site/blob/master/ecommerce/front/trackers.es6#L52)

File: front/js/shared/yandex.es6 (hunk 4 above)
Comment (artemiy312) [reply front/js/shared/yandex.es6]: @duker33 , yes, we publish onProductAdd event from both pages

File: front/js/shared/yandex.es6 (hunk 3 above)
Comment (duker33) [reply front/js/shared/yandex.es6]: @artemiy312 , so create plz pdd task or refarm-site's task with related comment here

File: front/js/shared/yandex.es6 (hunk 4 above)
Comment (duker33) [reply front/js/shared/yandex.es6]: @artemiy312 , code needs comment too :)

File: front/js/shared/yandex.es6 [hunk 6]
```diff
@@ -19,7 +19,7 @@
   };
 
   // @todo #129 Implement tracking of certain actions on front-end for YA and GA.
-  //  Actions: one-click purchase, changing products count on order page.
+  //  Actions: one-click purchase, changing products count on the onProductAddorder page.
```
Comment (duker33) [inline front/js/shared/yandex.es6]: `onProductAddorder` -> `onProductAddOrder`

File: front/js/shared/yandex.es6 [hunk 7]
```diff
@@ -19,6 +18,23 @@
...
+  //  See the parent issue for a detail.
+
+  // Sync container for yaTracker
+  window.dataLayer = window.dataLayer || [];
+  // Load ecommerce plugin for gaTracker
+  try {
+    ga('require', 'ecommerce');
+  } catch (e) {
+    let ga = console.error;
+    console.error(`GaTracker did not load. Traceback: ${e}`);
```
Comment (duker33) [inline front/js/shared/yandex.es6]: `GaTracker not loaded` or `GaTracker was not loaded`

**Pull request merged**